src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,256 @@
+                        ORDER_RECEIVED_GROUP_ERROR, ERROR_RECOVERY_OFFSET));
+                registry.getListenerContainer(ORDER_RECEIVED_GROUP_ERROR).pause();
+            }
         }
     }
 
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_ERROR, groupId = ORDER_RECEIVED_GROUP_ERROR,
-            autoStartup = "${uk.gov.companieshouse.item-handler.error-consumer}")
-    public void processOrderReceivedError(String orderReceivedUri) {
-        try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC_ERROR);
-        } catch (Exception x){
-            LOGGER.error("Message: " + orderReceivedUri + " received on topic: " + ORDER_RECEIVED_TOPIC_ERROR + " could not be processed.");
-        }
+    protected void logMessageReceived(org.springframework.messaging.Message<OrderReceived> message, String topic){
+        OrderReceived msg = message.getPayload();
+        LOGGER.info(String.format("'order-received' message: [orderUri: \"%1$s\", %2$s] received from topic: \"%3$s\".",
+                msg.getOrderUri(), getMessageHeadersAsString(message), topic));
     }
 
-    private ConsumerConfig getConsumerConfig() {
-        ConsumerConfig consumerConfig = new ConsumerConfig();
-        consumerConfig.setMaxRetries(MAX_RETRY_ATTEMPTS);
-        consumerConfig.setKeyDeserializer(StringDeserializer.class.getName());
-        consumerConfig.setValueDeserializer(StringDeserializer.class.getName());
-        consumerConfig.setResetOffset(false);
-        consumerConfig.setBrokerAddresses(brokerAddresses.split(","));
-        consumerConfig.setAutoCommit(true);
-        return consumerConfig;
-    }
-
-    private ProducerConfig getProducerConfig() {
-        ProducerConfig config = new ProducerConfig();
-        if (brokerAddresses != null && !brokerAddresses.isEmpty()) {
-            config.setBrokerAddresses(brokerAddresses.split(","));
-        } else {
-            throw new ProducerConfigException("Broker addresses for kafka broker missing, check if environment variable KAFKA_BROKER_ADDR is configured. " +
-                    "[Hint: The property 'spring.kafka.consumer.bootstrap-servers' uses the value of this environment variable in live environments " +
-                    "and that of 'spring.embedded.kafka.brokers' property in test.]");
-        }
+    protected void logMessageProcessingFailureRecoverable(org.springframework.messaging.Message<OrderReceived> message,
+                                                        String currentTopic, String nextTopic, Exception exception) {
+        OrderReceived msg = message.getPayload();
+        LOGGER.error(String.format("'order-received' message: [orderUri: \"%1$s\", %2$s] from topic: \"%3$s\" " +
+                        "has failed processing. Publishing to topic \"%4$s\" for recovery. Recoverable exception is - \n%4$s",
+                msg.getOrderUri(), getMessageHeadersAsString(message), currentTopic, nextTopic, exception.getStackTrace()));
```

Comment:
This is going to be a long message to read in the logs. This could make use of a data map to store the topic against a `data.topic` key and the offset against a `data.offset` key. A key `stack_trace` is also being used for stack traces. The keys could then be used for filtering the logs making it easier to trace the messages and identify errors. This would also be consistent with the logging output from other Kafka consumers.

Reply:
The current logging produces a message as -
`error: 'order-received' message: [orderUri: "/orders/12345", key: "order-received-retry", offset: 15, partition: 0] from topic: "order-received-retry" has failed processing. Publishing to topic "order-received-error" for recovery. Recoverable exception is - 
order-received-error`

I like the idea of a data map with traceable keys. I'll update the implementation.